cmdstanpy/cmdstan_args.py

```diff
@@ -0,0 +1,398 @@
+                raise ValueError(
+                    'sampling_iters must be a non-negative integer'.format(
+                        self.sampling_iters
+                    )
+                )
+
+        if self.thin is not None:
+            if self.thin < 1:
+                raise ValueError(
+                    'thin must be at least 1, found {}'.format(self.thin)
+                )
+
+        if self.max_treedepth is not None:
+            if self.max_treedepth < 1:
+                raise ValueError(
+                    'max_treedepth must be at least 1, found {}'.format(
+                        self.max_treedepth
+                    )
+                )
+
+        if self.step_size is not None:
+            if isinstance(self.step_size, (int, float)):
+                if self.step_size < 0:
+                    raise ValueError(
+                        'step_size must be > 0, found {}'.format(self.step_size)
+                    )
+            else:
+                if len(self.step_size) != chains:
+                    raise ValueError(
+                        'number of step_sizes must match number of chains '
+                        ' found {} step_sizes for {} chains '.format(
+                            len(self.step_size), chains
+                        )
+                    )
+                for i in range(len(self.step_size)):
+                    if self.step_size[i] < 0:
+                        raise ValueError(
+                            'step_size must be > 0, found {}'.format(
+                                self.step_size[i]
+                            )
+                        )
+
+        if self.metric is not None:
+            dims = None
+            if isinstance(self.metric, str):
+                if self.metric in ['diag', 'diag_e']:
+                    self.metric = 'diag_e'
+                elif self.metric in ['dense', 'dense_e']:
+                    self.metric = 'dense_e'
+                else:
+                    if not os.path.exists(self.metric):
+                        raise ValueError('no such file {}'.format(self.metric))
+                    dims = read_metric(self.metric)
+            elif isinstance(self.metric, list):
+                if len(self.metric) != chains:
+                    raise ValueError(
+                        'number of metric files must match number of chains '
+                        ' found {} metric files for {} chains '.format(
+                            len(self.metric), chains
+                        )
+                    )
+                names_set = set(self.metric)
+                if len(names_set) != len(self.metric):
+                    raise ValueError(
+                        'each chain must have its own metric file,'
+                        ' found duplicates in metric files list.'
+                    )
+                for i in range(len(self.metric)):
+                    if not os.path.exists(self.metric[i]):
```

Comment:
for i, metric in enumerate(self.metric):
        if not os.path.exists(metric):

Reply:
fixed